File: typescript/v1/tests/error-code.test.ts
import { describe, it, expect } from 'vitest';
import { ErrorCode, errorCodeFromCode } from '../src/error-code';

describe('errorCodeFromCode', () => {
  it('should map all defined codes', () => {
    expect(errorCodeFromCode('40001')).toBe(ErrorCode.INVALID_API_KEY);
    expect(errorCodeFromCode('40002')).toBe(ErrorCode.INVALID_SIGNATURE);
    expect(errorCodeFromCode('40003')).toBe(ErrorCode.REQUEST_EXPIRED);
    expect(errorCodeFromCode('40004')).toBe(ErrorCode.INVALID_PARAMETER);
    expect(errorCodeFromCode('40401')).toBe(ErrorCode.PAYMENT_NOT_FOUND);
    expect(errorCodeFromCode('40402')).toBe(ErrorCode.REFUND_NOT_FOUND);
    expect(errorCodeFromCode('40901')).toBe(ErrorCode.INVALID_PAYMENT_STATUS);
    expect(errorCodeFromCode('40902')).toBe(ErrorCode.DUPLICATE_REQUEST);
    expect(errorCodeFromCode('50001')).toBe(ErrorCode.NETWORK_ERROR);
    expect(errorCodeFromCode('50002')).toBe(ErrorCode.SERVER_ERROR);
    expect(errorCodeFromCode('50003')).toBe(ErrorCode.RATE_LIMITED);
    expect(errorCodeFromCode('59999')).toBe(ErrorCode.UNKNOWN_ERROR);
  });

  it('should return UNKNOWN_ERROR for unrecognized code', () => {
    expect(errorCodeFromCode('99999')).toBe(ErrorCode.UNKNOWN_ERROR);
    expect(errorCodeFromCode('')).toBe(ErrorCode.UNKNOWN_ERROR);
  });
});

describe('ErrorCode', () => {
  it('should have 12 error codes', () => {
    expect(Object.keys(ErrorCode).length).toBe(12);
  });
});

File: java/v1/src/test/java/com/paygate/sdk/ErrorCodeTest.java
package com.paygate.sdk;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class ErrorCodeTest {

    @Test
    void shouldMapAllDefinedCodes() {
        assertThat(ErrorCode.fromCode("40001")).isEqualTo(ErrorCode.INVALID_API_KEY);
        assertThat(ErrorCode.fromCode("40002")).isEqualTo(ErrorCode.INVALID_SIGNATURE);
        assertThat(ErrorCode.fromCode("40003")).isEqualTo(ErrorCode.REQUEST_EXPIRED);
        assertThat(ErrorCode.fromCode("40004")).isEqualTo(ErrorCode.INVALID_PARAMETER);
        assertThat(ErrorCode.fromCode("40401")).isEqualTo(ErrorCode.PAYMENT_NOT_FOUND);
        assertThat(ErrorCode.fromCode("40402")).isEqualTo(ErrorCode.REFUND_NOT_FOUND);
        assertThat(ErrorCode.fromCode("40901")).isEqualTo(ErrorCode.INVALID_PAYMENT_STATUS);
        assertThat(ErrorCode.fromCode("40902")).isEqualTo(ErrorCode.DUPLICATE_REQUEST);
        assertThat(ErrorCode.fromCode("50001")).isEqualTo(ErrorCode.NETWORK_ERROR);
        assertThat(ErrorCode.fromCode("50002")).isEqualTo(ErrorCode.SERVER_ERROR);
        assertThat(ErrorCode.fromCode("50003")).isEqualTo(ErrorCode.RATE_LIMITED);
        assertThat(ErrorCode.fromCode("59999")).isEqualTo(ErrorCode.UNKNOWN_ERROR);
    }

    @Test
    void shouldReturnUnknownForUnrecognizedCode() {
        assertThat(ErrorCode.fromCode("99999")).isEqualTo(ErrorCode.UNKNOWN_ERROR);
        assertThat(ErrorCode.fromCode("")).isEqualTo(ErrorCode.UNKNOWN_ERROR);
    }

    @Test
    void allCodesShouldBeAlignedWithSpec() {
        // Verify counts match spec/error-codes.yaml
        assertThat(ErrorCode.values()).hasSize(12);
    }
}
